hal/src/main/native/athena/REVPDH.cpp

```diff
@@ -89,6 +89,22 @@ void InitializeREVPDH() {
 
 extern "C" {
 
+/**
+ * REV PDH firmware v25.0.0 updates the scaling factor for channels 20-23.
+ */
+static double getLowCurrentChannelConversionFactor(HAL_REVPDHHandle* handle) {
+  HAL_PowerDistributionVersion version;
+  uint32_t status;
+  HAL_GetREVPDHVersion(handle, &version, &status);
```

Comment:
This will delay 100ms each call if the PDH has never been connected to CAN- it's an edge case but the status frame getters are generally expected to be nonblocking. Since this function is internal it could be passed the PDH struct and use the cached version directly.

Reply:
The issue with that is if the bus comes up too late, this will never work after that. 

We should probably have a "Non Blocking" version reader, that will check to see if the map has received the version packet, and if not send the RTR packet, not wait, and just assume one version. Then the next read will check, likely have the value, and then work correctly.

Also make sure we do a version check in the initializer too.

Reply:
But, if the device isn't on the bus, it will block every call as you said. However, if you switch to always read the cache, if the value isn't there initially, nothing will ever update the cache value.

Reply:
Nvm, I see what you mean- misread which cache you were referring to. That sounds like a good idea.

Reply:
Basically, I think the call needs to be this.

1. If cache exists, read from cache and return.
2. Check CAN packet buffer, see if the version packet exists there. If so, update cache and return.
2. If not cache exists, send RTR packet. Don't wait, return assuming (new/old, needs to be decided) fw version.

The next loop around would then be able to read out of the packet buffer.

The other option is a different API ID for that status packet. That additionally solves the issue, without needing a version check.